lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1125,6 +1096,11 @@ def run(self):
         return msg, changed, rebuilt, differences.get_legacy_docker_diffs(), facts
 
 
+def _detect_publish_mode_usage(client):
+    for publish_def in client.module.params.get('publish', []):
+        return 'mode' in publish_def.keys()
```

Comment:
Also, since `publish_def` is a `dict`, it's enough to write:
```suggestion
        return 'mode' in publish_def
```